### PR TITLE
feat(web): dedicated Profiles page with read-only lifecycle hooks

### LINKED ISSUE #950: Web UI editor for profiles and hooks
> _Disclaimer: this issue was drafted by Claude after a quick discussion with me — I reviewed and agree with the contents._

Today the TUI has a settings dialog where I can edit profiles and configure hooks (`on_create` / `on_launch` / `on_destroy`). On the web side there's no equivalent — if I want to create or tweak a profile I have to drop into the terminal and hand-edit TOML.

Given that the web UI is increasingly the primary surface for a lot of users, this feels like a real friction point — especially for the "create a workflow preset, pick it from the wizard" flow we discussed in #929. Without a web editor, the picker is read-only from the web's perspective.

### What I'm thinking

- A profiles/settings page in the web app, parity with the TUI's settings dialog
- CRUD for profiles: name, agent, sandbox, auto-approve, env, hooks
- Editor for the three hook lifecycles with the same string-or-array shape the TOML accepts
- Proper warnings/confirmations for hook-trust just like the TUI flow

### Open questions

- Should this also expose repo-level config (`.agent-of-empires/config.toml`) editing, or just profile-level? My instinct is **profile only** for v1 — repo config feels more "checked into the project" and editing it from the web UI is weirder.
- For hooks specifically, do we want a "test run" button so users can see what their hook does before committing? Probably nice-to-have, not v1.

### Related

Spun off from #929 — point 4 in [my comment](https://github.com/njbrake/agent-of-empires/issues/929#issuecomment-4395968167).

### PR DESCRIPTION
**Pull request by Seluj78**
## Description

Adds a dedicated **Profiles** page to the web dashboard (`/profiles`), reachable from a new sidebar-footer button. It pulls profile management out of the cramped Settings header dropdown into a master-detail page:

- **Left rail:** every profile with a `default` badge.
- **Detail panel:** create / rename / delete / set-default, inline description editing, and **Edit configuration** buttons that deep-link into `/settings/<tab>?profile=<name>` for per-section editing (the passphrase-gated sandbox/worktree saves stay in Settings, untouched).
- **Read-only lifecycle hooks:** On Create / On Launch / On Destroy, each labeled by source (profile override / overridden-to-none / inherited-global / none), with an explain-why note.

### Why hooks (and agent/env) are read-only

Setting a `hooks` section through the API runs arbitrary shell on session start, i.e. remote code execution. The server already blocks it (`ALLOWED_PROFILE_SETTINGS_SECTIONS`, pinned by a regression test), so this PR **respects that boundary**: hooks are displayed, never editable from the web, and the same applies to `agent_command_override` / `agent_extra_args` / `extra_env` / `custom_agents` / `agent_detect_as`. As defense in depth, the API client now mirrors the server allowlist (`PROFILE_WRITABLE_SECTIONS`) and refuses (loudly, no request) any PATCH carrying a blocked/unknown section.

Settings also learns to deep-link: `SettingsView` reads `?profile=` and keeps it in sync when the header dropdown changes, plus a request-id guard so a stale settings fetch can't clobber the current profile after a fast switch.

## PR Type

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording 

## Test Coverage Analysis

**Web dashboard / cockpit (Playwright user story)**
- [x] Added or updated a Playwright spec under `web/tests/` and updated `web/tests/coverage-matrix.json`

**TUI / CLI (e2e test)**
- [x] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle

## How I tested

Web-only change (no Rust touched), so the cockpit/cargo gates are N/A; ran the web suites locally:

- `tsc -b` clean, `eslint` clean on changed files.
- `vitest run`: 1383 passed (new: `api.test.ts` write-guard, `profileHooks.test.ts`, `HooksReadOnlyPanel.test.tsx`, `ProfilesPage.test.tsx`).
- `node tests/validate-coverage-matrix.mjs` passes.
- Live Playwright spec `tests/live/profiles-page.spec.ts` added (CRUD / set-default / description persistence / deep-link / read-only hooks invariant).

### User-runnable checks
- [ ] `aoe serve`, open the dashboard, click the Profiles icon in the sidebar footer → lands on `/profiles`.
- [ ] Create, rename, delete, and set-default a profile; reload and confirm each persists.
- [ ] Edit a profile description, reload, confirm it persists.
- [ ] Click an **Edit configuration** button → lands on `/settings/<tab>?profile=<name>` with that profile selected in the header.
- [ ] The hooks panel shows On Create / On Launch / On Destroy read-only with the explain-why note and no input controls.
- [ ] `aoe serve --read-only` hides the create/rename/delete/set-default/description controls but keeps the list and read-only hooks visible.

## AI Usage

- [x] AI was used

**AI Model/Tool used:** Claude Opus 4.8 (1M context) via Claude Code, with a 3-model design debate (Gemini 3.1 Pro, GPT-5.5, Claude Opus 4.7) to settle the architecture.

- [x] I am an AI Agent filling out this form (check box if true)

Closes #950

This PR adds a dedicated Profiles page and client-side safeguards so users can manage configuration profiles in the web dashboard while keeping unsafe sections read-only.

High-level changes
- New Profiles UI: Adds a master–detail Profiles page at /profiles, reachable from a new sidebar-footer button. Left rail lists profiles (with a default badge); right panel shows profile details, description editing, actions (create, rename, delete, set-default), deep-links to settings tabs, and a read-only lifecycle hooks section.
- Profile management flows: Create, rename, delete (with confirmation), set-as-default, and persist description changes; name validation enforces safe characters.
- Read-only unsafe sections: Lifecycle hooks (on_create, on_launch, on_destroy) and other sensitive fields (agent command/args/env, custom_agents, agent_detect_as) are shown but not editable; hooks are labeled by source (override / override-empty / inherited / none) with an explanatory note about remote code execution risk.
- URL & navigation sync: SettingsView preserves and syncs ?profile= across tabs and dropdowns; App routing and sidebar handlers support opening/closing /profiles. Tutorial auto-launch is suppressed on /profiles.
- Client-side write guard: API client exposes PROFILE_WRITABLE_SECTIONS and refuses PATCH requests containing blocked or unknown sections (logs and returns false) to prevent accidental unsafe updates.
- Read-only mode: `aoe serve --read-only` hides mutation controls while preserving listing and read-only hooks.
- Tests & docs: Vitest unit/integration tests and a Playwright e2e spec cover CRUD, set-default, description persistence, deep-linking, read-only hooks, and the client write-guard. Documentation updated and coverage matrix extended.

Benefits
- Makes profile management accessible without editing TOML.
- Reduces risk of remote code execution and misconfiguration by preventing web edits to unsafe sections and adding client-side validation.
- Improves navigation and discoverability via deep-links and a dedicated page.
- Makes hook provenance clear (why a command runs or is disabled).

Potential follow-ups
- Ensure server-side enforcement mirrors the client allowlist and returns clear errors (critical).
- Add an audited, sandboxed "test run" for hooks with explicit trust controls.
- Revisit which profile fields can be safely exposed for editing and add logging/auditing for profile mutations.
- Accessibility and UX polish for mobile/sidebar interactions and tutorial behavior.

Technical details (concise)
- New/changed components: ProfilesPage, HooksReadOnlyPanel; WorkspaceSidebar adds onProfiles; App.tsx routing supports /profiles; SettingsView gains profile prop/onSelectProfile and a request-id (loadSeq) guard.
- Hook handling: Added buildEffectiveHooks + types (HookEventKey, HookSource, EffectiveHookGroup) to merge profile/global hooks and report sources (override, override-empty, inherited, none).
- Types: Added HooksOverride and ProfileSettingsResponse.
- API client: getProfileSettings returns typed ProfileSettingsResponse; updateProfileSettings enforces PROFILE_WRITABLE_SECTIONS and blocks invalid PATCH bodies.
- Concurrency guards: loadSeq pattern used in SettingsView and ProfilesPage to avoid stale async results clobbering state.
- Tests: Vitest for hooks merging, HooksReadOnlyPanel, ProfilesPage, and api write-guard; Playwright spec for /profiles; coverage-matrix updated.

### CODE REVIEW

File: web/src/components/profiles/ProfilesPage.tsx [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+      ([profile, global]) => {
+        if (seq !== loadSeq.current) return;
+        setProfileSettings(profile);
+        setGlobalHooks(global?.hooks as HooksOverride | undefined);
+        setDescription(
+          typeof profile?.description === "string" ? profile.description : "",
+        );
+      },
+    );
+  }, [selected]);
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply web/src/components/profiles/ProfilesPage.tsx:97]: Added a seq-guarded `.catch` that resets state and surfaces an error. Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/1757/commits/0a5cd0d3eda0cf43ae37eb7b9b0a1dc6f42a9941

File: web/src/components/SettingsView.tsx [hunk 2]
```diff
@@ -173,16 +188,28 @@ export function SettingsView({
...
+  // Profiles page deep-links raise the odds of rapid profile changes.
+  const loadSeq = useRef(0);
   const loadSettings = useCallback(() => {
     if (!selectedProfile) return;
+    const seq = ++loadSeq.current;
     fetchSettings(selectedProfile).then((s) => {
+      if (seq !== loadSeq.current) return;
       if (s) setSettings(s);
     });
   }, [selectedProfile]);
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply web/src/components/SettingsView.tsx:220]: Added a seq-guarded `.catch` that clears settings on failure. Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/1757/commits/48e266285f88c45121f9ff5e2d5b995b650a59af

**Comment by Seluj78** [comment]
> **Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

> @coderabbitai (outside-diff, `web/src/App.tsx`): Exclude `/profiles` from tour auto-launch readiness.

Done: added `!showProfiles` to `tourAutoLaunchReady` so the tutorial no longer auto-opens on the Profiles route. Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/1757/commits/dfd1b9e09e6b66137754965952f2d5d4847bdd92

**Review by njbrake — CHANGES_REQUESTED** [review]
_Note: this review was drafted by Claude via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._

Nice feature. The architecture is sound, the read-only hooks boundary is the right security call and is well documented, and the test coverage is genuinely good. One regression is blocking, plus a note on the test that should have caught it.

## Blocker: the write-guard breaks saving Cockpit settings for every profile

`PROFILE_WRITABLE_SECTIONS` in `web/src/lib/api.ts` lists 10 sections but omits `cockpit`. The server's `ALLOWED_PROFILE_SETTINGS_SECTIONS` (`src/server/api/mod.rs:98`) has 11, including `cockpit` (the server strips only `node_path` via `COCKPIT_BLOCKED_FIELDS` and accepts the rest).

The Cockpit settings tab saves through this path:

```
CockpitSettings onSaveField("cockpit", ...)
  -> saveSubField -> saveField -> sendSave("cockpit", ...)
  -> updateProfileSettings(profile, { cockpit: {...} })
```

With this PR the PATCH now hits the new guard, `cockpit` is not in the set, so it logs an error and returns `false` without sending the request. The UI then shows "Failed to save, please try again". Before this PR `updateProfileSettings` had no guard and the save worked, so this is a regression: every Cockpit setting (tool durations, queue-drain mode, concurrent resumes, idle timeouts) becomes unsavable from the dashboard. The Cockpit tab is unconditionally present in `buildSidebar()`, so it is not edge-case gated.

The pin test (`api.test.ts:197`) hard-codes the same 10-element list, and its comment claims it "fails CI if the client list drifts; keep it in sync with the Rust constant." It enshrines the drift rather than catching it; there is no real cross-language check, so CI stays green while the feature is broken.

Fix:
- Add `cockpit` to `PROFILE_WRITABLE_SECTIONS` in `api.ts` (before `description`, to match server order).
- Add `cockpit` to the pinned list in `api.test.ts`.

## Follow-up suggestion (non-blocking)

The "keep this in sync by hand, no automated pin" comment is the footgun that just bit. Worth a follow-up issue to derive the client allowlist from the Rust constant via a single source (for example the `/api/about` payload or a generated JSON) so the two cannot drift silently.

## CodeRabbit findings

All three (unhandled promise rejections in the `loadSeq` effects of `ProfilesPage` and `SettingsView`, and the vacuous contract test) are already addressed in the current head. I agree with each and they are resolved.

## Verdict

Project fit: accept. This is a needed feature (closes #950), well scoped, with a correct security boundary and strong tests. The one thing before merge is the Cockpit allowlist gap, a two-line fix plus a one-line test update.

**Comment by Seluj78** [comment]
> **Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

Thanks @njbrake, good catch. Confirmed regression: `cockpit` was missing from the client `PROFILE_WRITABLE_SECTIONS`, so CockpitSettings saves got rejected by the new guard even though the server accepts the section.

Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/1757/commits/e8cdb1d58c3105b3988f57e78c9e29bc9b789089: added `cockpit` to the allowlist (between `logging` and `description`, matching `ALLOWED_PROFILE_SETTINGS_SECTIONS`) and to the pinned test.

On the follow-up: agreed the hand-synced allowlist is a footgun. I'll file an issue to single-source the client allowlist from the Rust constant (e.g. surface it on `/api/about` or a generated JSON) so they can't drift silently. CodeRabbit's three findings were already addressed at head.

**Review by njbrake — APPROVED** [review]
_Note: this review was drafted by Claude via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._

Re-reviewed at head. The Cockpit allowlist regression from my earlier review is resolved in commit e8cdb1d5: `cockpit` is now present in `PROFILE_WRITABLE_SECTIONS` (web/src/lib/api.ts, in server order before `description`), and the pin test in web/src/lib/api.test.ts was updated to match. Both allowlists are now identical 11-entry sets.

The read-only hooks boundary holds: `HooksReadOnlyPanel` exposes no inputs or save handlers and renders commands as text, so there is no mutation path or XSS surface, and the client write-guard refuses blocked sections before any fetch. Coverage across unit, vitest, and live Playwright is solid.

Approving and superseding my prior changes-requested. The hand-synced allowlist is still worth a follow-up to derive from a single source so the two cannot drift again.
